tests/lib/eslint/eslint.js

```diff
@@ -4885,6 +4885,73 @@ describe("ESLint", () => {
+         * @returns {boolean} `true` if the number is a float, `false` otherwise.
+         */
+        function isNumber(n) {
+            return Number(n) === n && n % 1 !== 0;
+        }
+
+        it("should report stats", async () => {
+            const engine = new ESLint({
+                overrideConfigFile: true,
+                overrideConfig: {
+                    rules: {
+                        "no-regex-spaces": "error"
+                    }
+                },
+                cwd: getFixturePath("stats-example"),
+                stats: true
+            });
+            const results = await engine.lintFiles(["file-to-fix.js"]);
+
+            assert.strictEqual(results[0].stats.fixPasses, 0);
+            assert.strictEqual(results[0].stats.times.passes.length, 1);
+            assert.strictEqual(isNumber(results[0].stats.times.passes[0].parse.total), true);
+            assert.strictEqual(isNumber(results[0].stats.times.passes[0].rules["no-regex-spaces"].total), true);
+            assert.strictEqual(isNumber(results[0].stats.times.passes[0].rules["wrap-regex"].total), true);
+            assert.strictEqual(results[0].stats.times.passes[0].fix.total, 0);
+            assert.strictEqual(isNumber(results[0].stats.times.passes[0].total), true);
+        });
+
+        it("should report stats with fix", async () => {
+            const engine = new ESLint({
+                overrideConfigFile: true,
+                overrideConfig: {
+                    rules: {
+                        "no-regex-spaces": "error"
+                    }
+                },
+                cwd: getFixturePath("stats-example"),
+                fix: true,
+                stats: true
+            });
+            const results = await engine.lintFiles(["file-to-fix.js"]);
+
+            assert.strictEqual(results[0].stats.fixPasses, 2);
+            assert.strictEqual(results[0].stats.times.passes.length, 3);
+            assert.strictEqual(isNumber(results[0].stats.times.passes[0].parse.total), true);
+            assert.strictEqual(isNumber(results[0].stats.times.passes[1].parse.total), true);
+            assert.strictEqual(isNumber(results[0].stats.times.passes[0].rules["no-regex-spaces"].total), true);
+            assert.strictEqual(isNumber(results[0].stats.times.passes[0].rules["wrap-regex"].total), true);
+            assert.strictEqual(isNumber(results[0].stats.times.passes[1].rules["no-regex-spaces"].total), true);
+            assert.strictEqual(isNumber(results[0].stats.times.passes[1].rules["wrap-regex"].total), true);
+            assert.strictEqual(isNumber(results[0].stats.times.passes[0].fix.total), true);
+            assert.strictEqual(isNumber(results[0].stats.times.passes[1].fix.total), true);
+            assert.strictEqual(isNumber(results[0].stats.times.passes[0].total), true);
+            assert.strictEqual(isNumber(results[0].stats.times.passes[1].total), true);
+        });
+
+    });
```

Comment:
Can you also add some tests with [processors](https://eslint.org/docs/latest/extend/custom-processors)? I just tried `--stats` with [`eslint-plugin-markdown`](https://github.com/eslint/eslint-plugin-markdown) and the result doesn't seem to contain any parsing and rules timings data. Perhaps the filename-based logic doesn't account for this case where we're linting virtual child files.

Reply:
@mdjermanovic: I've just added a test for processors. 👍

Doing so, I noticed that the lint times were only contributed to the results when using `lintFiles()` but not with `lintText()`. This was the result of not using the fully resolved filePath (for [placeholders](https://github.com/eslint/eslint/pull/17850/commits/439a7a75234a8f7b548567a90531c41150e8140a#diff-cb681134e1d583c87b8a91a2dfbd4de0bfe5dd84fa3ee57efb5028d7bfeb5219R494)) in my `getTimes()` helper function. to retrieve the respective `times` object.

Could you please try again and let me know whether you are still missing any timings data?